pkg/util/users/users.go

```diff
@@ -152,6 +158,25 @@ func (strategy DefaultUserSyncStrategy) ExecuteSyncRequests(requests []spec.PgSy
 	return nil
 }
 
+func resolveOwnerMembership(dbOwner spec.PgUser, additionalOwners []string, db *sql.DB) error {
+	errors := make([]string, 0)
+	for _, groupRole := range dbOwner.MemberOf {
+		for _, additionalOwner := range additionalOwners {
+			if additionalOwner == groupRole {
```

Comment:
I don't quite get the logic here. So we want to go from the current buggy setup
```
 db_owner               |                           | {}
 cron_admin             | Cannot login              | {db_owner}
 ```
to the one with the corrected membership. For that part of the code that means simply revoking the wrong role as in 
```
 db_owner               |                           | {}
 cron_admin             | Cannot login              | {}
 ```
For that simple example, the `dbOwner.MemberOf` will be empty, and the  `REVOKE` won't get executed ?

maybe the vars are mis-named ?

Reply:
actually the `for _, groupRole := range dbOwner.MemberOf {` loop is not needed. Whenever there are additional owner roles configured the operator should try to revoke db owners from them